Small Code improvement

Sigmanificient opened the pull request:
Added Typing and minor code change to improve readability.

Sigmanificient committed:
removing useless import and variables

Sigmanificient committed:
improving code style and removed useless imports

Sigmanificient committed:
simplifying command_controller function

Sigmanificient committed:
improved overall code

Sigmanificient committed:
improved overall code

Sigmanificient committed:
fully Typing and small improvements

Sigmanificient committed:
fully Typing and small improvements

Sigmanificient committed:
Merge remote-tracking branch 'origin/main' into main

# Conflicts:
#	utils/imageViewer.py

Muhimen123 commented on utils/downloader.py:
```diff
@@ -35,30 +39,28 @@ def download_content(link: str, path: str, file_name: str):
-            while not progress.finished:
-                progress.update(task, advance=chunk_size)
+        while not progress.finished:
+            progress.update(task, advance=chunk_size)
```
task not defined.

Sigmanificient replied:
fixed !

Muhimen123 commented on utils/downloader.py:
```diff
@@ -35,30 +39,28 @@ def download_content(link: str, path: str, file_name: str):
 
-        with Progress() as progress:
-            download_task = progress.add_task(f"[green] {file_name}", total=image_size)
+    with Progress() as progress:
```
Images not downloading properly. Try putting the second context manger(for progress) in the first context manager(res) because second context manager can't access the image_content properly since res is a stream.

Sigmanificient replied:
The issue has been fixed ! I dispatched the progress into it own function to ensure code flatness

Muhimen123 commented on utils/imageSaver.py:
```diff
@@ -29,22 +30,19 @@ def save_image():
-
-        path = os.path.join(os.getcwd(), "data", "images")
+        path: str = os.path.join(os.getcwd(), "data", "images")
         downloader.download_image(link, path, "current_desktop_wallpaper.png")
```
Instead of downloading `current_desktop_wallpaper.png` in `images` folder, it's saving it in a file name like `imagescurrent_desktop_wallpaper.png`. It's because the path doesn't end with `//`. Prefixing the `current_desktop_wallpaper.png` with `//` should work.

Sigmanificient replied:
fixed !

Muhimen123 commented on utils/imageSaver.py:
```diff
@@ -29,22 +30,19 @@ def save_image():
         downloader.download_image(link, path, "current_desktop_wallpaper.png")
-        wallpaper_path = path + "current_desktop_wallpaper.png"
-    
+        wallpaper_path: str = f"{path}current_desktop_wallpaper.png"
```
Same problem here as well.

Sigmanificient replied:
fixed !

Muhimen123 reviewed (changes requested):
Hi! Thanks for the type hinting in the code! But as I mentioned earlier, I found few(minor) bugs in the code. It will be a great help if you fix them up! Other than that, I don't see any problem in the code so far. 

Thanks again, have a  good day!

Sigmanificient committed:
- changed module name to snake case
[fix] task not defined in downloader

Sigmanificient committed:
[fix] download progress and image saving

Sigmanificient committed:
[fix] image saveing path

Sigmanificient committed:
[fix] image saveing path

Sigmanificient committed:
typo

Sigmanificient committed:
refactoring download prompt
[fix] get_new_file_name verification

Muhimen123 commented on utils/image_saver.py:
```diff
@@ -0,0 +1,144 @@
+
+        path: str = "//" + os.path.join(os.getcwd(), "data", "images")
+        downloader.download_image(link, path, "current_desktop_wallpaper.png")
+        wallpaper_path: str = f"//{path}current_desktop_wallpaper.png"
```
Looks like you misplaced `//` here. Should it go in between `{path}` and `curren...`

Muhimen123 reviewed (changes requested):
Can you fix the bug I mentioned earlier? Also, looks like you forgot the put one extra blank line at the end of the file. Can you do that(no big deal if you can't) since it makes pep8 happy. 

Thanks for the previous bug fix though. No issue there.

Sigmanificient committed:
Update image_saver.py

Move // at right position
- pep8 blank is already present but not showing on github preview

Muhimen123 commented:
Did you test the code? For some reason, saving wallpaper isn't working for. You can take a look if you are free. I am currently trying to find out what went wrong in image save command.

Muhimen123 commented on utils/downloader.py:
```diff
@@ -35,61 +53,71 @@ def download_content(link: str, path: str, file_name: str):
+    chunk_size: int = 1024
+    image_content: Iterator = res.iter_content(chunk_size)
+
+    file_path: str = path + file_name
```
can you replace line 64 with this line instead? 
```python
file_path: str = os.path.join(path, file_name)
```

Muhimen123 commented on utils/image_saver.py:
```diff
@@ -0,0 +1,144 @@
+            "https://raw.githubusercontent.com/Muhimen123/TID/main/image_data.json"
+        ) if answer == 'TID Repo' else input("Please enter download link: ")
+
+        path: str = "//" + os.path.join(os.getcwd(), "data", "images")
```
Also, please remove the prefixing `//`

Muhimen123 reviewed (changes requested):
This changes should probably fix the problems. 😄

Sigmanificient committed:
[fix] save wallpaper working

Muhimen123 reviewed (approved):
Thanks for you patience. Everything working fine so far. Will let you know if anything breaks.
